src/types/message.ts

```diff
@@ -196,13 +196,20 @@ export interface Application {
   bot?: UserPayload;
 }
 
+/** https://discord.com/developers/docs/resources/channel#message-object-message-reference-structure */
 export interface Reference {
-  /** The id of the originating message */
-  "message_id"?: string;
-  /** The id of the originating message's channel */
-  "channel_id": string;
-  /** The id of the originating message's guild */
-  "guild_id"?: string;
+  /** ID of the originating message */
+  message_id?: string;
+  /**
+   * ID of the originating message's channel
+   * 
+   * NOTE: `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model
+   */
+  channel_id?: string;
+  /** ID of the originating message's guild */
+  guild_id?: string;
+  /** When sending, whether to error if the referenced message doesn't exist instead of sending as a normal (non-reply) message, default true */
+  fail_if_not_exists?: boolean;
```

Comment:
Please make this default to false in the code! We should not allows bots to fail and crash!

Reply:
keep in mind if you set it to false here you won't be able to set it to true.

Reply:
> keep in mind if you set it to false here you won't be able to set it to true.
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reply:
No i meant in the code, the function itself that handles this.

```ts
{
...options,
fail_if_not_exists: options.failIfNotExists === true
}
```

Reply:
What? That's equivalent of passing it directly